neo4j/driver_with_context_test.go

```diff
@@ -0,0 +1,507 @@
+			expectedErr:           fmt.Errorf("in summary: nope"),
+		},
+		{
+			description: "returns error when read result summary cannot be retrieved",
+			configurers: []ExecuteQueryConfigurationOption{WithReadersRouting()},
+			createSession: &fakeSession{
+				executeReadTransactionResult: &fakeResult{
+					keys:       keys,
+					collect:    records,
+					summaryErr: fmt.Errorf("in summary: nope"),
+				}},
+			expectedSessionConfig: defaultSessionConfig,
+			expectedErr:           fmt.Errorf("in summary: nope"),
+		},
+		{
+			description: "returns error when read execution fails",
+			configurers: []ExecuteQueryConfigurationOption{WithReadersRouting()},
+			createSession: &fakeSession{
+				executeReadErr: fmt.Errorf("oopsie"),
+			},
+			expectedSessionConfig: defaultSessionConfig,
+			expectedErr:           fmt.Errorf("oopsie"),
+		},
+	}
+
+	for _, testCase := range testCases {
+		outer.Run(testCase.description, func(t *testing.T) {
+			driver := &driverWithContext{
+				newSession: func(_ context.Context, config SessionConfig) SessionWithContext {
+					AssertDeepEquals(t, testCase.expectedSessionConfig, config)
+					return testCase.createSession
+				},
+				defaultManagedSessionBookmarkManager: defaultBookmarkManager,
+			}
+
+			eagerResult, err := driver.ExecuteQuery(ctx, "RETURN 42", nil, testCase.configurers...)
+
+			AssertDeepEquals(t, testCase.expectedErr, err)
+			AssertDeepEquals(t, testCase.expectedResult, eagerResult)
+		})
+	}
+
+	outer.Run("default bookmark manager is thread-safe", func(t *testing.T) {
+		driver := &driverWithContext{
+			newSession: func(_ context.Context, config SessionConfig) SessionWithContext {
+				return &fakeSession{
+					executeWriteErr: fmt.Errorf("oopsie, write failed"),
+				}
+			},
+		}
+
+		var wait sync.WaitGroup
+		var bookmarkManagerAddresses sync.Map
+		goroutineCount := 50
+		wait.Add(goroutineCount)
+		for i := 0; i < goroutineCount; i++ {
+			go func(i int) {
+				callExecuteQueryOrBookmarkManagerGetter(driver, i)
+				storeBookmarkManagerAddress(
+					&bookmarkManagerAddresses,
+					driver.defaultManagedSessionBookmarkManager.(*bookmarkManager))
+				wait.Done()
+			}(i)
+		}
+		wait.Wait()
+
+		addressCounts := make(map[uintptr]int32, goroutineCount)
+		bookmarkManagerAddresses.Range(func(key, value any) bool {
+			addressCounts[key.(uintptr)] = *(value.(*int32))
+			return true
+		})
+		if len(addressCounts) != 1 {
+			t.Errorf("expected exactly 1 bookmark manager pointer to have been created, got %v", addressCounts)
+		}
+		address := uintptr(unsafe.Pointer(driver.defaultManagedSessionBookmarkManager.(*bookmarkManager)))
+		if count, found := addressCounts[address]; !found || count != int32(goroutineCount) {
+			t.Errorf("expected pointer address %v to be seen %d time(s), got these instead %v", address, count, addressCounts)
+		}
+	})
+}
+
+func callExecuteQueryOrBookmarkManagerGetter(driver *driverWithContext, i int) {
+	if i%2 == 0 {
```

Comment:
```suggestion
	if i % 2 == 0 {
```
maybe? don't know much about Go style guide but looked off to me.

Reply:
go fmt is king here :)